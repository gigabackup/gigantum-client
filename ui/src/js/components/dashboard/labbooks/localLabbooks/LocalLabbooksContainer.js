// vendor
import {
  createFragmentContainer,
  graphql,
} from 'react-relay';
// components
import Labbooks from '../Labbooks';

export default createFragmentContainer(
  Labbooks,
  {
    labbookList: graphql`
      fragment LocalLabbooksContainer_labbookList on LabbookQuery{
        labbookList{
          id
          ...LocalLabbooks_localLabbooks
        }
      }
    `,
  },
);
