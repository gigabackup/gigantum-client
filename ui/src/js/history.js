import { createBrowserHistory } from 'history';

/**
*  @param {}
*  Mehtod gets basename for history object
*  @return {String}
*/
const getBasename = () => {
  const { pathname } = document.location;
  const pathList = pathname.split('/');
  const uniqueClientString = (pathList.length > 2)
    ? pathList[2]
    : '';
  const basename = (process.env.BUILD_TYPE === 'cloud')
    ? `/run/${uniqueClientString}/`
    : '/';
  return basename;
};

const basename = getBasename();

export default createBrowserHistory({
  basename,
  forceRefresh: false,
});
