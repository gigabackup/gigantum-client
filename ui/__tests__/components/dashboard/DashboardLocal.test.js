import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';
import history from 'JS/history';
import { mount } from 'enzyme';
import LocalLabbooksContainer from 'Pages/dashboard/labbooks/localLabbooks/LocalLabbooksContainer';
import store from 'JS/redux/store';
import { BrowserRouter as Router } from 'react-router-dom';

import relayTestingUtils from '@gigantum/relay-testing-utils';
import json from './__relaydata__/DashboardLocal.json';


const fixtures = {
  auth: () => {},
  localLabbooks: json.data.labbookList,
  labbookList: json.data.labbookList,
  history: {
    replace: () => jest.fn(),
    location: {
      hash: '',
    },
  },
  refetchSort: () => {},
  loading: false,
};


describe('DashboardLocal', () => {
  test('Renders a snapshot', () => {
    const wrapper = renderer.create(
        relayTestingUtils.relayWrap(
          <Provider store={store}>
            <Router>
              <LocalLabbooksContainer {...fixtures} />
            </Router>
          </Provider>,
         {}, json.data,
       ),
    );

    const tree = wrapper.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
