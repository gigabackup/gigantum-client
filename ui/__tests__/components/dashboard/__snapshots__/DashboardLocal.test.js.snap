// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardLocal Renders a snapshot 1`] = `
<div
  className="Labbooks"
>
  <div />
  <div
    className="Labbooks__panel-bar"
  >
    <h6
      className="Labbooks__username"
    >
      uitest
    </h6>
    <h2
      className="Labbooks__title"
    >
      Projects
    </h2>
  </div>
  <div
    className="Labbooks__menu  mui-container flex-0-0-auto"
  >
    <ul
      className="Labbooks__nav  flex flex--row"
    >
      <li
        className="Labbooks__nav-item Labbooks__nav-item--local Labbooks__nav-item--selected"
      >
        <a
          onClick={[Function]}
        >
          Local
        </a>
      </li>
      <li
        className="Labbooks__nav-item Labbooks__nav-item--cloud"
      >
        <a
          onClick={[Function]}
        >
          Cloud
        </a>
      </li>
      <hr
        className="Labbooks__navigation-slider Labbooks__navigation-slider--local"
      />
      <div
        className="hidden cloudLocal"
      >
        <div
          className="ToolTip__toggle cloudLocal"
          onClick={[Function]}
        >
          <div
            className="ToolTip__glow-container"
          >
            <div
              className="ToolTip__glow-ring-outer"
            >
              <div
                className="ToolTip__glow-ring-inner"
              />
            </div>
          </div>
        </div>
        <div
          className="hidden cloudLocal"
        />
        <div
          className="hidden cloudLocal"
        >
          This allows you to switch between Local and Cloud Project listings. Both Projects you have created and Projects that have been shared with you will be visible.
        </div>
      </div>
    </ul>
  </div>
  <div
    className="Labbooks__subheader grid"
  >
    <div
      className="Labbooks__search-container column-2-span-6 padding--0"
    >
      <input
        className="Labbooks__search no--margin"
        defaultValue=""
        onFocus={[Function]}
        onKeyUp={[Function]}
        placeholder="Filter Projects by name or description"
        type="text"
      />
    </div>
    <div
      className="LabbookFilterBy column-4-span-3 padding--0"
    >
      Filter by:
      <span
        className="LabbookFilterBy__selector Dropdown Dropdown--collapsed"
        onClick={[Function]}
      >
        All
      </span>
      <ul
        className="LabbookFilterBy__menu Dropdown__menu box-shadow hidden"
      >
        <li
          className="LabbookFilterBy__list-item"
          onClick={[Function]}
        >
          All 
          ✓ 
        </li>
        <li
          className="LabbookFilterBy__list-item"
          onClick={[Function]}
        >
          My Projects 
          
        </li>
        <li
          className="LabbookFilterBy__list-item"
          onClick={[Function]}
        >
          Shared with me 
          
        </li>
      </ul>
    </div>
    <div
      className="LabbookSort column-4-span-3 padding--0"
    >
      Sort by:
      <span
        className="LabbookSort__selector Dropdown Dropdown--collapsed"
        onClick={[Function]}
      >
        Modified Date (Newest)
      </span>
      <ul
        className="LabbookSort__menu Dropdown__menu box-shadow hidden"
      >
        <li
          className="LabbookSort__list-item"
          onClick={[Function]}
        >
          Modified Date (Newest) 
          ✓ 
        </li>
        <li
          className="LabbookSort__list-item"
          onClick={[Function]}
        >
          Modified Date (Oldest) 
          
        </li>
        <li
          className="LabbookSort__list-item"
          onClick={[Function]}
        >
          Creation Date (Newest) 
          
        </li>
        <li
          className="LabbookSort__list-item"
          onClick={[Function]}
        >
          Creation Date (Oldest) 
          
        </li>
        <li
          className="LabbookSort__list-item"
          onClick={[Function]}
        >
          A-Z 
          
        </li>
        <li
          className="LabbookSort__list-item"
          onClick={[Function]}
        >
          Z-A 
          
        </li>
      </ul>
    </div>
  </div>
  <div
    className="Labbooks__listing"
  >
    <div
      className="grid"
    >
      <div
        className="ImportModule Card Card--line-50 Card--text-center Card--add Card--import column-4-span-3"
      >
        <div
          className="Import__labbook-main"
        >
          <div
            className="Import__labbook-header"
          >
            <div
              className="Import__labbook-icon"
            >
              <div
                className="Import__labbook-add-icon"
              />
            </div>
            <div
              className="Import__labbook-title"
            >
              <h4>
                Add Project
              </h4>
            </div>
          </div>
          <div
            className="btn--import"
            onClick={[Function]}
          >
            Create New
          </div>
          <div
            className="btn--import"
            onClick={[Function]}
          >
            Import Existing
          </div>
          <div
            className="hidden createLabbook"
          >
            <div
              className="ToolTip__toggle createLabbook"
              onClick={[Function]}
            >
              <div
                className="ToolTip__glow-container"
              >
                <div
                  className="ToolTip__glow-ring-outer"
                >
                  <div
                    className="ToolTip__glow-ring-inner"
                  />
                </div>
              </div>
            </div>
            <div
              className="hidden createLabbook"
            />
            <div
              className="hidden createLabbook"
            >
              Clicking on the 'Create New' button will open the new Project window, letting you name and select a Base for the new Project.
            </div>
          </div>
        </div>
        <div
          className="hidden"
        />
      </div>
      <a
        className="Card Card--300 Card--text column-4-span-3 flex flex--column justify--space-between"
        href="/projects/uitest/ui-test-remote"
        onClick={[Function]}
      >
        <div
          className="LocalLabbooks__row--icons"
        >
          <div
            className="LocalLabbooks__containerStatus"
          >
            <div
              className="ContainerStatus__container-state ContainerStatus__containerStatus--state loading box-shadow"
              data-tooltip="Rebuild Required, container will attempt to rebuild before starting."
              onClick={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <div
                className="ContainerStatus__text"
              >
                
              </div>
              <div
                className="ContainerStatus__toggle"
              >
                <div
                  className="ContainerStatus__toggle-btn"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="LocalLabbooks__row--text"
        >
          <div>
            <h6
              className="LocalLabbooks__panel-title"
              onClick={[Function]}
            >
              <span>
                <span
                  className=""
                >
                  ui-test-remote
                </span>
              </span>
            </h6>
          </div>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner "
          >
            uitest
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner"
          >
            Created on Fri Jan 18 2019
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner"
          >
            Modified Fri Jan 18 2019
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--description"
          >
            <span
              className="LocalLabbooks__description--blank"
            >
              No description provided
            </span>
          </p>
        </div>
        <div
          className="Tooltip-Listing LocalLabbookPanel__loading"
          data-tooltip="loading"
        />
      </a>
      <a
        className="Card Card--300 Card--text column-4-span-3 flex flex--column justify--space-between"
        href="/projects/uitest/ui-test-project"
        onClick={[Function]}
      >
        <div
          className="LocalLabbooks__row--icons"
        >
          <div
            className="LocalLabbooks__containerStatus"
          >
            <div
              className="ContainerStatus__container-state ContainerStatus__containerStatus--state loading box-shadow"
              data-tooltip="Rebuild Required, container will attempt to rebuild before starting."
              onClick={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <div
                className="ContainerStatus__text"
              >
                
              </div>
              <div
                className="ContainerStatus__toggle"
              >
                <div
                  className="ContainerStatus__toggle-btn"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="LocalLabbooks__row--text"
        >
          <div>
            <h6
              className="LocalLabbooks__panel-title"
              onClick={[Function]}
            >
              <span>
                <span
                  className=""
                >
                  ui-test-project
                </span>
              </span>
            </h6>
          </div>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner "
          >
            uitest
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner"
          >
            Created on Fri Jan 18 2019
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner"
          >
            Modified Fri Jan 18 2019
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--description"
          >
            <span>
              <span
                className=""
              >
                This is a ui test project

              </span>
            </span>
          </p>
        </div>
        <div
          className="Tooltip-Listing LocalLabbookPanel__loading"
          data-tooltip="loading"
        />
      </a>
      <a
        className="Card Card--300 Card--text column-4-span-3 flex flex--column justify--space-between"
        href="/projects/uitest/test-tt"
        onClick={[Function]}
      >
        <div
          className="LocalLabbooks__row--icons"
        >
          <div
            className="LocalLabbooks__containerStatus"
          >
            <div
              className="ContainerStatus__container-state ContainerStatus__containerStatus--state loading box-shadow"
              data-tooltip="Rebuild Required, container will attempt to rebuild before starting."
              onClick={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <div
                className="ContainerStatus__text"
              >
                
              </div>
              <div
                className="ContainerStatus__toggle"
              >
                <div
                  className="ContainerStatus__toggle-btn"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="LocalLabbooks__row--text"
        >
          <div>
            <h6
              className="LocalLabbooks__panel-title"
              onClick={[Function]}
            >
              <span>
                <span
                  className=""
                >
                  test-tt
                </span>
              </span>
            </h6>
          </div>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner "
          >
            uitest
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner"
          >
            Created on Fri Jan 11 2019
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner"
          >
            Modified Fri Jan 11 2019
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--description"
          >
            <span
              className="LocalLabbooks__description--blank"
            >
              No description provided
            </span>
          </p>
        </div>
        <div
          className="Tooltip-Listing LocalLabbookPanel__loading"
          data-tooltip="loading"
        />
      </a>
      <a
        className="Card Card--300 Card--text column-4-span-3 flex flex--column justify--space-between"
        href="/projects/uitest/new-project"
        onClick={[Function]}
      >
        <div
          className="LocalLabbooks__row--icons"
        >
          <div
            className="LocalLabbooks__containerStatus"
          >
            <div
              className="ContainerStatus__container-state ContainerStatus__containerStatus--state loading box-shadow"
              data-tooltip="Rebuild Required, container will attempt to rebuild before starting."
              onClick={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <div
                className="ContainerStatus__text"
              >
                
              </div>
              <div
                className="ContainerStatus__toggle"
              >
                <div
                  className="ContainerStatus__toggle-btn"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="LocalLabbooks__row--text"
        >
          <div>
            <h6
              className="LocalLabbooks__panel-title"
              onClick={[Function]}
            >
              <span>
                <span
                  className=""
                >
                  new-project
                </span>
              </span>
            </h6>
          </div>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner "
          >
            uitest
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner"
          >
            Created on Wed Jan 16 2019
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner"
          >
            Modified Thu Jan 24 2019
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--description"
          >
            <span
              className="LocalLabbooks__description--blank"
            >
              No description provided
            </span>
          </p>
        </div>
        <div
          className="Tooltip-Listing LocalLabbookPanel__loading"
          data-tooltip="loading"
        />
      </a>
      <a
        className="Card Card--300 Card--text column-4-span-3 flex flex--column justify--space-between"
        href="/projects/uitest/awful-intersections-demo"
        onClick={[Function]}
      >
        <div
          className="LocalLabbooks__row--icons"
        >
          <div
            className="LocalLabbooks__containerStatus"
          >
            <div
              className="ContainerStatus__container-state ContainerStatus__containerStatus--state loading box-shadow"
              data-tooltip="Rebuild Required, container will attempt to rebuild before starting."
              onClick={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <div
                className="ContainerStatus__text"
              >
                
              </div>
              <div
                className="ContainerStatus__toggle"
              >
                <div
                  className="ContainerStatus__toggle-btn"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="LocalLabbooks__row--text"
        >
          <div>
            <h6
              className="LocalLabbooks__panel-title"
              onClick={[Function]}
            >
              <span>
                <span
                  className=""
                >
                  awful-intersections-demo
                </span>
              </span>
            </h6>
          </div>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner "
          >
            uitest
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner"
          >
            Created on Thu May 24 2018
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner"
          >
            Modified Fri Jan 11 2019
          </p>
          <p
            className="LocalLabbooks__paragraph LocalLabbooks__paragraph--description"
          >
            <span>
              <span
                className=""
              >
                A fun demo LabBook to automatically find and generate cringe-worthy maps of horrible intersections
              </span>
            </span>
          </p>
        </div>
        <div
          className="Tooltip-Listing LocalLabbookPanel__loading"
          data-tooltip="loading"
        />
      </a>
      <div
        className="LabbookPaginationLoader--hidden"
      >
        <div />
      </div>
      <div
        className="LabbookPaginationLoader--hidden"
      >
        <div />
      </div>
      <div
        className="LabbookPaginationLoader--hidden"
      >
        <div />
      </div>
      <div
        className="LabbookPaginationLoader--hidden"
      >
        <div />
      </div>
      <div
        className="LabbookPaginationLoader--hidden"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
